app/src/main/java/com/github/orkest/View/sharing/SharingActivity.kt

```diff
@@ -63,35 +88,124 @@ class SharingComposeActivity : ComponentActivity() {
         }
     }
 
+    /**
+     * This function is used to connect to the Spotify API and get the song name
+     *
+     */
+    private fun spotifyAuthorization(){
```

Comment:
Maybe you could try to do some dependency injection to allow yourself to test the correct retrieval of a song, I know it is a hard method to test but maybe it could be an idea to try